File: CSharpCodeBase/core/objectpool.lua
local Class = require 'utils.hump.class'

local ObjectPool = Class({init=function(self, class)
      self._class = class
      self._poolLength = 1
      self._curPoint = 1
      self._pool = {}
      self:Init()
end})

function ObjectPool:Init()
  for i = 1, self._poolLength do
    if self._pool[i] == nil then
      self._pool[i] = self._class()
      self._pool[i].__pool = self
    end
  end
end

function ObjectPool:Fetch(...)
  if self._curPoint > self._poolLength then
    self._poolLength = self._poolLength + 1
    self:Init()
  end
  local object = self._pool[self._curPoint]
  self._curPoint = self._curPoint + 1
  if object.OnFetch then object:OnFetch(...) end
  return object
end

function ObjectPool:Release(object)
  self._curPoint = self._curPoint - 1
  self._pool[self._curPoint] = object
  if object.OnRelease then object:OnRelease() end
end

return ObjectPool

File: CSharpCodeBase/core/objectpool.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;

namespace MainGame
{
    public class objectpool
    {
        public object _class;

        public void objectpool(object class_name)
        {
            this._class = class_name;
            this._poolLength = 1;
            this._curPoint = 1;
            this._pool = new object[_poolLength];
            this.Init();
        }
        public void Init()
        {
            for (int i = 0; i < this._poolLength; i++)
            {
                if (this._pool[i] == null)
                {
                    //this._pool[i] = this._class();
                    //this._pool[i].__pool = self;
                }
            }
        }

        public void Fetch(params object param)
        {
            if (this._curPoint > this._poolLength)
            {
                this._poolLength = this._poolLength + 1;
                this.Init();
            }
            else
            {
                var object_s = this._pool[this._curPoint];
                this._curPoint = this._curPoint + 1;
                //if(object_s.OnFetch  ){  object_s.OnFetch(); }
                //return object;
            }
        }

        public int _poolLength { get; set; }
        public int _curPoint { get; set; }
        public object[] _pool;

        public void Release(object objects)
        {
            this._curPoint = this._curPoint - 1;
            this._pool[this._curPoint] = objects;
            //if(object.OnRelease  ){  object:OnRelease() }
        }
    }
}